tx/src/entities.rs

```diff
@@ -69,12 +66,30 @@ pub struct LookupRef {
     pub v: edn::Value, // An atom.
 }
 
+/// A "transaction function" that exposes some value determined by the current transaction.  The
+/// prototypical example is the current transaction ID, `(current-tx)`.
+///
+/// A natural next step might be to expose the current transaction instant `(current-txInstant)`,
+/// but that's more difficult: the transaction itself can set the transaction instant (with some
+/// restrictions), so the transaction function must be late-binding.  Right now, that's difficult to
```

Comment:
Perhaps you can work around that by simply defining `(current-txInstant)` as the timestamp that the transactor would use _by default_.

That would allow callers to do fun things like impose a timestamp but keep the transactor's timestamp around in a separate property, or correct clock drift, or whatever.

You could define the transactor's behavior as having a default

```edn
[:db/add (current-tx) :db/txInstant (current-txInstant)]
```

… which 'feels' better if you call it `(current-instant)` or `(transact-instant)`.

Reply:
I think, in fact, that `(transaction-tx)` and `(transaction-instant)` might be better than any reference to "current", since current might be reasonably interpreted to mean "the latest transaction committed" or "the timestamp when I built this data structure".  I find that `(current-txInstant)` blurs that considerably.

As for allowing an eager `(current-txInstant)` while allowing to set `:db/txInstant`... it just seems so prone to error.  Follow-up, for sure.